lib/couchbase3.js

```diff
@@ -140,3 +145,159 @@ Couchbase.prototype.disconnect = function(callback) {
   this.cluster = null;
   return promise.then(resolved, callback);
 };
+
+/**
+ * Create a new model instance for the given data
+ * @param {String} model The model name
+ * @param {Object} data The model data
+ * @param {Function} [callback] The callback function
+ */
+Couchbase.prototype.create = function (model, data, callback) {
+  var self = this;
+  debug("Couchbase.CREATE: " + JSON.stringify([model, data]));
+
+  if (callback == null) {
+    callback = noop;
+  }
+
+  // Handle with promise.
+  var promise;
+  var resolved = function(res) {
+    return callback(null, res);
+  };
+
+  // The cached promise.
+  promise = this._connection;
+  if (promise == null) {
+    debug('No connections. Try to connect...');
+    self.connect(callback);
+    promise = this._connection;
+  }
+
+  //Data
+  var id = self.getIdValue(model, data);
+
+  // Insert.
+  promise = promise.then(function(bucket) {
+    bucket.upsert(id, data,function(err, result) {
+      if (err) {
+        debug('create failed');
+        return callback(err,null);
```

Comment:
This is not how you handle errors. See https://github.com/petkaantonov/bluebird/blob/master/API.md#catchfunction-errorclassfunction-predicate-function-handler---promise

Also please check (by looking into node-couchbase source code for example) if the `err` here is an instance of `Error` and if not please wrap it here (with tools like https://github.com/Wiredcraft/build-http-error for example).